File: project-nexus-portal/fe/src/services/matchmakingService.js
import api from './api';

export const matchmakingService = {
  getDashboard: async () => {
    const response = await api.get('/matchmaking/dashboard');
    return response.data;
  },

  getPosts: async (params = {}) => {
    const response = await api.get('/matchmaking/posts', { params });
    return response.data;
  },

  getPostById: async (id) => {
    const response = await api.get(`/matchmaking/posts/${id}`);
    return response.data;
  },

  createPost: async (postData) => {
    const response = await api.post('/matchmaking/posts', postData);
    return response.data;
  },

  closePost: async (id) => {
    const response = await api.patch(`/matchmaking/posts/${id}/close`);
    return response.data;
  },

  getCandidates: async (params = {}) => {
    const response = await api.get('/matchmaking/candidates', { params });
    return response.data;
  },

  submitJoinRequest: async (postId, data) => {
    const response = await api.post(`/matchmaking/posts/${postId}/join-requests`, data);
    return response.data;
  },

  getPostJoinRequests: async (postId) => {
    const response = await api.get(`/matchmaking/posts/${postId}/join-requests`);
    return response.data;
  },

  getMyJoinRequests: async () => {
    const response = await api.get('/matchmaking/join-requests/my-requests');
    return response.data;
  },

  reviewJoinRequest: async (requestId, decision) => {
    const response = await api.put(`/matchmaking/join-requests/${requestId}/review`, null, {
      params: { decision },
    });
    return response.data;
  },
};

export default matchmakingService;

File: project-nexus-portal/be/src/main/java/com/nexus/portal/service/MatchmakingService.java
package com.nexus.portal.service;

import com.nexus.portal.dto.request.MatchmakingPostCreateRequest;
import com.nexus.portal.dto.request.TeamJoinRequestCreateRequest;
import com.nexus.portal.dto.response.CandidateResponse;
import com.nexus.portal.dto.response.MatchmakingDashboardResponse;
import com.nexus.portal.dto.response.MatchmakingPostResponse;
import com.nexus.portal.dto.response.TeamJoinRequestResponse;
import com.nexus.portal.enums.PostType;
import com.nexus.portal.enums.RequestStatus;

import java.util.List;

public interface MatchmakingService {

    MatchmakingDashboardResponse getDashboard(String currentUserEmail);

    List<MatchmakingPostResponse> getPosts(PostType type, Long facultyId, Long cohortId, Long departmentId, String query);

    MatchmakingPostResponse getPostById(Long id);

    MatchmakingPostResponse createPost(MatchmakingPostCreateRequest request, String currentUserEmail);

    void closePost(Long id, String currentUserEmail);

    List<CandidateResponse> getCandidates(Long facultyId, Long cohortId, Long majorId, String query, String currentUserEmail);

    TeamJoinRequestResponse submitJoinRequest(Long postId, TeamJoinRequestCreateRequest request, String currentUserEmail);

    TeamJoinRequestResponse reviewJoinRequest(Long requestId, RequestStatus decision, String reviewerEmail);

    List<TeamJoinRequestResponse> getMyJoinRequests(String currentUserEmail);

    List<TeamJoinRequestResponse> getPostJoinRequests(Long postId, String currentUserEmail);
}
